backend/src/main/java/com/google/models/RelatedData.java

```diff
@@ -0,0 +1,157 @@
+import java.util.HashMap;
+import java.util.TreeMap;
+import java.util.SortedMap;
+import java.time.format.DateTimeParseException;
+
+/** 
+ * Store related data for anomalies with data points and meta data.
+ * This is an immutable class. 
+ */
+public final class RelatedData {
+  public static final String RELATED_DATA_ENTITY_KIND = "relatedData";
+  public static final String USERNAME_PROPERTY = "username";
+  public static final String METRIC_NAME_PROPERTY = "metricName";
+  public static final String DIMENSION_NAME_PROPERTY = "dimensionName";
+  public static final String DATA_POINTS_PROPERTY = "dataPoints";
+
+  private static final String INDENT = "  ";
+  private static final String INDENT_DOUBLE = "    ";
+  private static final String DUMMY_USERNAME = "bob@";
+  private static final String DUMMY_METRIC_NAME = "Sample metric name";
+  private static final String DUMMY_DIMENSION_NAME = "Sample dimension name";
+  private static final Map<Timestamp, MetricValue> DUMMY_DATA_POINTS = ImmutableMap.of( 
+      Timestamp.getDummyTimestamp(1), new MetricValue(1), 
+      Timestamp.getDummyTimestamp(2), new MetricValue(2), 
+      Timestamp.getDummyTimestamp(3), new MetricValue(3));
+  
+  private final String metricName;
+  private final String dimensionName;
+  private final Map<Timestamp, MetricValue> dataPoints;
+  private final String username;
+
+  public RelatedData(String username, String metricName, String dimensionName, 
+      Map<Timestamp, MetricValue> dataPoints) {
+    this.username = username;
+    this.metricName = metricName;
+    this.dimensionName = dimensionName;
+    this.dataPoints = ImmutableMap.copyOf(dataPoints);
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public String getMetricName() {
+    return metricName;
+  }
+
+  public String getDimensionName() {
+    return dimensionName;
+  }
+
+  public Map<Timestamp, MetricValue> getDataPoints() {
+    return dataPoints;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+
+    if (!(o instanceof RelatedData)) {
+      return false;
+    }
+
+    RelatedData target = (RelatedData) o;
+
+    return target.metricName.equals(metricName)
+        && target.dimensionName.equals(dimensionName) 
+        && target.username.equals(username)
+        && dataPoints.entrySet().stream().allMatch(
+          e -> e.getValue().equals(target.dataPoints.get(e.getKey()))
+        );
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder str = new StringBuilder("");
+    str.append(INDENT + "Username: " + username + "\n");
+    str.append(INDENT + "Metric Name: " + metricName + "\n");
+    str.append(INDENT + "Dimension Name: " + dimensionName + "\n");
+    str.append(INDENT + "Datapoints: \n");
+    dataPoints.forEach((key, value) -> 
+      str.append(INDENT_DOUBLE + key + ": " + value + "\n")
+    );
+    return str.toString();
+  }
+
+  public static RelatedData getDummyRelatedData() {
```

Comment:
qq why is this part of the `RelatedData` class?

Reply:
oh i had reread your code and for some reason I thought 'DummyRelatedData' is a separate class, and I was wondering why you didnt just have a static method in `DummyRelatedData` that returns `RelatedData`. But turns out that's not what the code is doing.

Can you add a comment to the dummy related data public static method?

Reply:
For sure! I'll add the comment to clarify this.